core/controllers/base.py

```diff
@@ -401,7 +404,9 @@ def _render_exception(self, error_code, values):
         values['code'] = error_code
 
         # This checks if the response should be JSON or HTML.
-        if self.payload is not None:
+        # For get method, the payload is None so GET_HANDLER_TYPE takes effect
```

Comment:
Minor suggested edit (for flow): For GET requests, there is no payload, so we check against GET_HANDLER_TYPE. Otherwise, we check whether self.payload exists.